### PR TITLE
Fix module `path_type` creation when globals does not contain `__name__`

### PR DESCRIPTION
**Pull request by hramezani**
Fixes https://github.com/pydantic/pydantic/issues/8448

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by hramezani** [comment]
please review

File: pydantic/type_adapter.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def __init__(
         except AttributeError:
             if module is None:
                 f = sys._getframe(1)
-                module = cast(str, f.f_globals['__name__'])
+                module = cast(str, f.f_globals.get('__name__', ''))
```
Comment (sydney-runkle) [inline pydantic/type_adapter.py:207]: Do we want to default to `'__main__'` instead of `''`?

**Review by sydney-runkle — APPROVED** [review]
Looks great @hramezani, thanks!

File: pydantic/type_adapter.py (hunk 1 above)
Comment (hramezani) [reply pydantic/type_adapter.py:207]: hmm, not sure. @samuelcolvin do you have any opinion here?

**Comment by sydney-runkle** [comment]
Going ahead and merging this. Super easy fix if we want to change the default from `''` to `'__main__'` down the line. As long as we make a decision before the next release, I think we're set.

**Pull request merged**